### PR TITLE
Explicit methods in DI Registry

### PR DESCRIPTION
**Pull request by nupplaphil**
- Replaces all `@method static` annotations with concrete methods
- Split the methods based on their namespace in different Registry (sub-)classes for better finding
- Rework some tests, which weren't working the "right way"
- Fixing some places in the code where `Cache` namespace was missing

I searched per regex and patterns and PHPStorm native functionallity for every occurrence of `DI` and tested it locally and on my friendica node ..

Needs https://github.com/friendica/friendica-addons/pull/941 as well

@annando with this PR, VSC should be working with `DI::app()` or `Core::config()` methods again and show the right hints. Would it be possible to pull this branch and check if I'm right?

### CODE REVIEW

**Comment by AlfredSK** [comment]
Jeez! You broke GitHub. 😂
I get an error when I try to switch to the changes tab 'page needs way too long to load'.

File: src/Registry/DI.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Friendica\Registry;
+
+use Dice\Dice;
+use Friendica\App as A;
```
Comment (MrPetovan) [inline src/Registry/DI.php]: Alias is superfluous here.

**Comment by MrPetovan** [comment]
Tests are failing.

**Comment by nupplaphil** [comment]
> Not sure about the single-letter namespace aliases. It looks weird at first glance and is probably harder to grep for. I'd prefer longer terms, like explicit `StorageRepository` instead of `S` for example.

I wasn't sure either .. Currently, I do have a deja-vú .. There was a similar situation with one-letter namespaces before, wasn't it? .. Anyway, I'll rename :-)

**Comment by nupplaphil** [comment]
> Not sure about the single-letter namespace aliases. It looks weird at first glance and is probably harder to grep for. I'd prefer longer terms, like explicit `StorageRepository` instead of `S` for example.

What about the Registry usages in the code, where `App` is already imported and I use `A` for the registry App. I'm not sure if `RegistryApp` won't be too long. Shall I really rename it?

**Comment by MrPetovan** [comment]
It can't never be "too long", but I just found out it can be "too short". And I would name it `AppRegistry`. Namespace is less precise to more precise, but when writing aliases I like it to be the opposite. `AppRegistry`, `EntityFactory`, 'ContactModel`, etc...

**Comment by nupplaphil** [comment]
> It can't never be "too long", but I just found out it can be "too short". And I would name it `AppRegistry`. Namespace is less precise to more precise, but when writing aliases I like it to be the opposite. `AppRegistry`, `EntityFactory`, 'ContactModel`, etc...

OK, as far as I know, I thought @annando hates typing long class-names, so I saw it as a service having as short classes as possible ^^

**Comment by MrPetovan** [comment]
Fair enough, I'd like him to chime in about what he would be more comfortable with. My IDE can handle one-letter alias names.

**Comment by nupplaphil** [comment]
@mrpetovan ans @annando I'd like to merge this and the add-on PR because:
- every other PR merge probably leads to a rebase of this one
- Fixes the https://github.com/friendica/friendica/issues/7998#issuecomment-573290305 
- Fixes other missing use statements as well

**Comment by annando** [comment]
I'd like to delegate the merging to @MrPetovan, since I cannot judge about this high level stuff. By brain most likely got stuck at "level 1" of class coding - and this whole stuff here is "level 99".

Concerning the names I like to have them not too long but also speaking - and unique across the whole code.

I hadn't had a look at how VSCode is handling this, since yesterday had been a busy day - and I don't know how to fetch the code.

**Comment by nupplaphil** [comment]
Ok, not too long but speaking: `use Repository\App as RApp;` for conflicting class names?

**Comment by nupplaphil** [comment]
uhoh .. I created this branch on the wrong repository, closing in favor of "right" repository

**Pull request closed without merging**